lib/StaticNarrative/exporter/generic_set_navigator.py

```diff
@@ -207,12 +216,42 @@ def _populate_set_item_info(self, set_list):
 
         return set_list
 
-    def _populate_set_item_ref_path(self, set_list):
+    def _populate_set_item_ref_path(
+        self: "GenericSetNavigator", set_list: list[dict[str, Any]]
+    ) -> list[dict[str, Any]]:
         for s in set_list:
             obj_spec = util.build_ws_obj_selector(s[REF], s.get("ref_path_to_set", []))
             util.populate_item_object_ref_paths(s["items"], obj_spec)
 
         return set_list
 
-    def _build_obj_ref(self, obj_info):
+    def _build_obj_ref(self: "GenericSetNavigator", obj_info: list[str | Any]) -> str:
         return str(obj_info[6]) + "/" + str(obj_info[0]) + "/" + str(obj_info[4])
+
+    def _get_ws_types(self: "GenericSetNavigator") -> list[str]:
```

Comment:
for an exciting upcoming extension to this package -- getting ready for a brave new world where we acknowledge that there are more set types than just `KBaseSets.ReadsSets`!

Reply:
😬